### PR TITLE
Simplify internals of domain models for page localization

### PR DESCRIPTION
**Pull request by alexisgrow**
TEST=manual,unit

### CODE REVIEW

File: src/commands/build/globalconfiglocalizer.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 const GlobalConfig = require('../../models/globalconfig');
 
 /**
- * GlobalConfigLocalizer is responsible for generating a set of localized
- * {@link GlobalConfig}s.
+ * GlobalConfigLocalizer is responsible for generating a localized
+ * {@link GlobalConfig}.
```
Comment (tmeyer2115) [inline src/commands/build/globalconfiglocalizer.js:5]: Minor nit: can fit on same line.

File: src/commands/build/pageconfigdecorator.js [hunk 2]
```diff
@@ -24,67 +19,65 @@ module.exports = class PageConfigDecorator {
...
   /**
-   * Creates a localized PageConfig for every page and locale, merging the rawConfigs
-   * based on the fallbacks and locale configuration provided.
+   * Creates an object mapping locale to a collection of decorated PageConfigs
+   * ({@link PageConfig}). These decorated PageConfigs each have all of the
+   * configuration for a given (page, locale) combination, including config
+   * based on the locale fallbacks.
    *
    * @param {Array<PageConfig>} pageConfigs
    * @returns {Array<PageConfig>}
```
Comment (tmeyer2115) [inline src/commands/build/pageconfigdecorator.js:28]: Does this return type need to be updated?

File: src/commands/build/globalconfiglocalizer.js (hunk 1 above)
Comment (alexisgrow) [reply src/commands/build/globalconfiglocalizer.js:5]: Updated!

File: src/commands/build/pageconfigdecorator.js (hunk 2 above)
Comment (alexisgrow) [reply src/commands/build/pageconfigdecorator.js:28]: Yes good catch!

**Pull request merged**